File: 004_Median_Of_Two_Sorted_Arrays/solution.py
class Solution(object):
    def findKth(self, nums1, nums2, k):
        len1 = len(nums1)
        len2 = len(nums2)
        if len1 > len2:
            return self.findKth(nums2, nums1, k)
        if len1 == 0:
            return nums2[k - 1]
        if k == 1:
            return min(nums1[0], nums2[0])
        p1 = min(k/2, len1)
        p2 = k - p1
        if nums1[p1 - 1] <= nums2[p2 - 1]:
            return self.findKth(nums1[p1:], nums2, p2)
        else:
            return self.findKth(nums1, nums2[p2:], p1)
        
        
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        len1 = len(nums1)
        len2 = len(nums2)
        if (len1 + len2) % 2 == 1:
            return self.findKth(nums1, nums2, (len1 + len2)/2 + 1)
        else:
            return (self.findKth(nums1, nums2, (len1 + len2)/2) + self.findKth(nums1, nums2, (len1 + len2)/2 + 1)) * 0.5
        

File: 004_Median_Of_Two_Sorted_Arrays/solution.go
func findMedianSortedArrays(nums1 []int, nums2 []int) float64 {
    i, j := 0, 0
    nums3 := []int{}
    for i < len(nums1) && j < len(nums2) {
        if nums1[i] < nums2[j] {
            nums3 = append(nums3, nums1[i])
            i = i + 1
        } else {
            nums3 = append(nums3, nums2[j])
            j = j + 1
        }
    }
    if i < len(nums1) {
        for ; i < len(nums1); i++ {
            nums3 = append(nums3, nums1[i])
        }
    }
    if j < len(nums2) {
        for ; j < len(nums2); j++ {
            nums3 = append(nums3, nums2[j])
        }
    }
    lenSum := len(nums3)
    if lenSum % 2 == 0 {
        return (float64(nums3[lenSum / 2 - 1]) + float64(nums3[lenSum / 2])) / 2.0
    } else {
        return float64(nums3[(lenSum + 1) / 2 - 1])
    }
}
